From 0bd627de56f28063d9fcef455587d0b7d11a00a6 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Fri, 2 Dec 2016 21:29:51 +0000
Subject: [PATCH] update-kernel: make mksquashfs exit on error

mksquashfs will ignore errors and continue by default by design. This is
definitively not the behavior we want so we tell it to explicitly exit
on error.
---
 update-kernel.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/update-kernel.in b/update-kernel.in
index 3e147c8..6d24649 100644
--- a/update-kernel.in
+++ b/update-kernel.in
@@ -259,7 +259,7 @@ find $ROOT/lib/modules -type f -name "*.ko" | xargs modinfo -F firmware | sort -
 		install -pD $ROOT/lib/firmware/$FW $MODLOOP/modules/firmware/$FW
 	fi
 done
-_exec mksquashfs $MODLOOP "$STAGING/$MODIMG" -comp xz
+_exec mksquashfs $MODLOOP "$STAGING/$MODIMG" -comp xz -exit-on-error
 
 _exec mkinitfs $MKINITFS_ARGS -q -b $ROOT -F "$features base squashfs" \
 	-o "$STAGING/initramfs-$FLAVOR" "$KVER"
-- 
2.11.0

